shopfloor/actions/stock.py

```diff
@@ -10,6 +10,36 @@ class StockAction(Component):
     _inherit = "shopfloor.process.action"
     _usage = "stock"
 
+    def is_src_location_valid(self, process, location):
```

Comment:
I would pass `picking_types` instead of process.
Alternatively you could move these to the process mixin but I think here is better because we can use them everywhere.

Reply:
@simahawk I did it on purpose to allow to add any other condition in the process to validate source location

Reply:
to do so, you should add this method to the service instead, and use this one by default, eg:
```py
[...service class...]
def is_src_location_valid(self, location):
    return self._action_for("stock").is_src_location_valid(self.picking_types, location)
```
then if you need customization for the specific service, you override it.